EcoSphereNetwork/CyberSpace
# Project Setup and Initial Implementation

This PR includes:

- Basic project structure setup
- TypeScript configuration
- Theme system implementation
- Core components (App, SceneManager)
- Type definitions for shaders and third-party modules
- Basic styling and layout
- Documentation updates
- Test setup

Changes:
- Added missing dependencies
- Fixed type errors
- Added theme support
- Created basic app structure
- Added documentation
- Set up testing environment
- Added development scripts

To test:
1. Clone the repository
2. Run `npm install`
3. Run `npm run dev`
4. Open http://localhost:5173 in your browser

You should see:
- CyberSpace logo
- 3D scene
- Basic UI elements

Interaction:
- Left mouse button: Rotate
- Right mouse button: Pan
- Mouse wheel: Zoom
- Double click: Focus on object
